File: db4on/reference/db4o-reference-chapters/src/com/db4odoc/f1/constructors/C4.cs
/* Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com */
using System;
using com.db4o;

namespace com.db4odoc.f1.constructors
{
	class C4 
	{
		private String s;
		[Transient] private int i;

		private C4(String s) 
		{
			this.s=s;
			this.i=s.Length;
		}

		override public String ToString() 
		{
			return s+i;
		}
	}
}

File: db4on/reference/vb/db4o-reference-chapters/src/com/db4odoc/f1/constructors/C4.vb
' Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com 
Imports System

Namespace com.db4odoc.f1.constructors
    <Serializable()> Class C4
        Private s As String
        <NonSerialized()> Private i As Integer

        Private Sub New(ByVal s As String)
            Me.s = s
            Me.i = s.Length
        End Sub

        Public Overrides Function ToString() As String
            Return s + i.ToString()
        End Function
    End Class
End Namespace
